src/ser/error.rs

```diff
@@ -7,7 +7,7 @@ use crate::bson::Bson;
 /// Possible errors that can arise during encoding.
 #[derive(Debug)]
 #[non_exhaustive]
-pub enum EncoderError {
+pub enum Error {
```

Comment:
Ditto above suggestion to prefer the name of the Error/Result types in this module